src/pytorch_lightning/core/saving.py

```diff
@@ -34,11 +33,12 @@
 from pytorch_lightning.utilities.migration import pl_legacy_patch
 from pytorch_lightning.utilities.parsing import parse_class_init_keys
 from pytorch_lightning.utilities.rank_zero import rank_zero_warn
-from pytorch_lightning.utilities.types import _PATH
+from pytorch_lightning.utilities.types import _DEVICE, _PATH
 
 log = logging.getLogger(__name__)
 PRIMITIVE_TYPES = (bool, int, float, str)
 ALLOWED_CONFIG_TYPES = (AttributeDict, MutableMapping, Namespace)
+MAP_LOCATION_TYPE = Optional[Union[_DEVICE, Callable[[_DEVICE], _DEVICE], Dict[_DEVICE, _DEVICE]]]
```

Comment:
can you keep this inside `utilities/types.py`?
I think there are more places where this can be used.

Reply:
and make it protected.

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13932/commits/f105d7f9663f02363a5c4774339bc9fa96fb75d3